AdminServer/appscale/admin/utils.py

```diff
@@ -204,44 +202,41 @@ def is_version_config(path):
     config_file_name = 'app.yaml'
 
     def is_version_config(path):
-      return canonical_path(path) == os.path.join(app_path, config_file_name)
+      return canonical_path(path, app_path) == os.path.join(app_path, config_file_name)
+
+  with tarfile.open(location, 'r:gz') as archive:
```

Comment:
I think tarfile.open can raise quite a few exception: should they be caught and re-raise as InvalidSource or something?